infrastructure-playbooks/upgrade-rhel7-to-rhel8.yml

```diff
@@ -0,0 +1,171 @@
+      debug:
+        msg: "{{ lookup('file', leapp_data_filename) }}"
+
+
+- hosts:
+    - "{{ mon_group_name | default('mons') }}"
+    - "{{ mgr_group_name | default('mgrs') }}"
+    - "{{ osd_group_name | default('osds') }}"
+    - "{{ mds_group_name | default('mdss') }}"
+    - "{{ rgw_group_name | default('rgws') }}"
+    - "{{ nfs_group_name | default('nfss') }}"
+    - "{{ rbdmirror_group_name | default('rbdmirrors') }}"
+    - "{{ iscsi_gw_group_name | default('iscsigws') }}"
+    - "{{ client_group_name | default('clients') }}"
+  become: true
+  tasks:
+    - import_role:
+        name: ceph-defaults
+
+    - name: enable repos
+      rhsm_repository:
+        name: ['rhel-7-server-rpms', 'rhel-7-server-extras-rpms']
+        purge: True
+
+    - name: set release
+      command: subscription-manager release --set {{ leapp_rhel_release }}
+      changed_when: false
+
+    - name: update system
+      command: yum update -y
+      changed_when: false
+
+
+- hosts:
+    - "{{ mon_group_name | default('mons') }}"
+    - "{{ mgr_group_name | default('mgrs') }}"
+    - "{{ osd_group_name | default('osds') }}"
+    - "{{ mds_group_name | default('mdss') }}"
+    - "{{ rgw_group_name | default('rgws') }}"
+    - "{{ nfs_group_name | default('nfss') }}"
+    - "{{ rbdmirror_group_name | default('rbdmirrors') }}"
+    - "{{ iscsi_gw_group_name | default('iscsigws') }}"
+  serial: 1
+  become: true
+  tasks:
+    - import_role:
+        name: ceph-defaults
+
+    - name: reboot              # if docker-to-podman was run before this playbook, the daemon won't come back.
+      reboot:                   # should we assume as soon as we started this playbook against that node we don't care if it doesn't come back here?
+        reboot_timeout: 600
+        test_command: uptime
+
+
+- hosts: "{{ client_group_name | default('clients') }}"
+  become: true
+  tasks:
+    - import_role:
+        name: ceph-defaults
+
+    - name: reboot
+      reboot:
+        reboot_timeout: 600
+        test_command: uptime
+
+
+- hosts: 
+    - "{{ mon_group_name | default('mons') }}"
+    - "{{ mgr_group_name | default('mgrs') }}"
+    - "{{ osd_group_name | default('osds') }}"
+    - "{{ mds_group_name | default('mdss') }}"
+    - "{{ rgw_group_name | default('rgws') }}"
+    - "{{ nfs_group_name | default('nfss') }}"
+    - "{{ rbdmirror_group_name | default('rbdmirrors') }}"
+    - "{{ iscsi_gw_group_name | default('iscsigws') }}"
+    - "{{ client_group_name | default('clients') }}"
+
+    - import_role:
+        name: ceph-defaults
+
+    - name: install leapp
+      package:
+        name: leapp
+        state: present
+
+    - name: untar leapp tarball
+      unarchive:
+        src: leapp-data8.tar.gz
+        dest: /etc/leapp/files
+      changed_when: false
+
+    - name: set PermitRootLogin   # if we start dealing with 'inhibitors' where's the limit? (either we deal with all of them or none)
```

Comment:
taking a look at https://github.com/oamg/leapp-repository/tree/master/repos/system_upgrade/el7toel8/actors

that's impossible to deal with all of this.